sign.go

```diff
@@ -200,3 +200,8 @@ func (s *Signature) DecodeNotDistinguishedFromEd25519(in [SignatureSize]byte) er
 	cp[63] |= 128
 	return s.Decode(cp)
 }
+
+// Equal returns true if the two signatures are equal
+func (s *Signature) Equal(other *Signature) bool {
```

Comment:
Is this being usee anywhere? I like it, but thinking through if it is needed or not

Reply:
I see that it's used in the tests. If it's only used in tests I think you can make it a test helper instead